decred/decred/dcr/rpc.py

```diff
@@ -80,6 +81,233 @@ def call(self, method, *params):
             raise Exception("%s error: %r" % (method, res["error"]))
         return res["result"]
 
+    def addNode(self, addr, subCmd):
+        """
+        Attempts to add or remove a persistent peer.
+
+        Args:
+            addr (str) IP address and port of the peer to operate on
+            subCmd (str) 'add' to add a persistent peer, 'remove' to remove a
+                persistent peer, or 'onetry' to try a single connection to a peer
+        """
+        self.call("addnode", addr, subCmd)
+
+    def createRawSSRTx(self, ticket, relayFee=None):
+        """
+        Returns a new unsigned transaction revoking the ticket.
+
+        Args:
+            ticket (msgTx.MsgTx): The ticket to revoke.
+            fee (int): Optional. Default=None. The fee to apply to the
+                revocation in atoms/kb.
+
+        Returns:
+            MstTx.msgtx: The revocation.
+        """
+        if relayFee:
+            outs = []
+            # Parse the ticket purchase transaction to determine the required output
+            # destinations for vote rewards or revocations.
+            (
+                ticketPayKinds,
+                ticketHash160s,
+                ticketValues,
+                _,
+                _,
+                _,
+            ) = txscript.sstxStakeOutputInfo(ticket.txOut)
+
+            # Calculate the output values for the revocation.  Revocations do not
+            # contain any subsidy.
+            values = txscript.calculateRewards(ticketValues, ticket.txOut[0].value, 0)
+
+            # All remaining outputs pay to the output destinations and amounts tagged
+            # by the ticket purchase.
+            for i in range(len(ticketHash160s)):
+                scriptFn = txscript.payToSSRtxPKHDirect
+                # P2SH
+                if ticketPayKinds[i]:
+                    scriptFn = txscript.payToSSRtxSHDirect
+                script = scriptFn(ticketHash160s[i])
+                outs.append(TxOut(values[i], script))
+
+            # Calculate the estimated signed serialize size.
+            scriptSizes = [txscript.RedeemP2SHSigScriptSize]
+            sizeEstimate = txscript.estimateSerializeSize(scriptSizes, outs, 0)
+            fee = txscript.calcMinRequiredTxRelayFee(relayFee, sizeEstimate)
+            # dcrd takes amounts in coins
+            fee /= 1e8
```

Comment:
Even though this is probably safe, let's be in the habit of rounding values when converting to and from atoms. 

`fee = round(fee / 1e8, 8)`

Here and a couple other places.